tools/wpt/browser.py

```diff
@@ -431,8 +431,55 @@ class Servo(Browser):
     product = "servo"
     requirements = "requirements_servo.txt"
 
+    def platform_string(self):
+        platform = {
+            "Linux": "linux",
+            "Windows": "win",
+            "Darwin": "mac"
+        }.get(uname[0])
+
+        if platform is None:
+            raise ValueError("Unable to construct a valid Servo package for current platform")
+
+        if platform == "linux":
+            bits = "-%s" % uname[4]
+            extension = ".tar.gz"
+        elif platform == "win":
+            bits = "64" if uname[4] == "x86_64" else "32"
+            extension = ".msi"
+        else:
+            bits = ""
+            extension = ".dmg"
+
+        return "%s" % (platform)
+
+    def find_extension(self, platform):
+        if platform == "linux":
+            extension = ".tar.gz"
+        elif platform == "win":
+            extension = ".msi"
+        else:
+            extension = ".dmg"
+
+        return "%s" % (extension)
+
     def install(self, dest=None):
-        raise NotImplementedError
+        """Install latest Browser Engine."""
+        if dest is None:
+            dest = os.pwd
+
+        extension = self.find_extension(self.platform_string())
+        url = "https://download.servo.org/nightly/%s/servo-latest%s" % (self.platform_string(), extension)
+
+        if format == "zip":
```

Comment:
Where does this come from?

Reply:
If I use the same method to return tuple of platform and extension, i cannot use them separately in the generating url which is needed here.